src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,131 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static StorageDao storageDao;
+    private static User testUser;
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+        testUser = new User();
+    }
+
+    @BeforeEach
+    void setUp() {
+        testUser.setLogin("logins78");
+        testUser.setPassword("password");
+        testUser.setAge(38);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        User actual = testUser;
+        actual.setAge(null);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(actual));
+    }
+
+    @Test
+    void register_lessMinAge_notOk() {
+        User actual = testUser;
+        actual.setAge(8);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(actual));
```

Comment:
17 - edge case have to be tested
![image](https://github.com/mate-academy/jv-registration-validation/assets/20326858/ea4ad042-951f-4ab4-970b-78fc61a87fea)